src/main/java/edu/unc/lib/boxc/migration/cdm/services/RedirectMappingIndexService.java

```diff
@@ -97,15 +100,50 @@ private void assertCollectionSubmitted() {
 
     public Connection openDbConnection() throws SQLException {
         try {
-            // TODO in BXC-3372 need to connect it to actual DB when not in local testing
-            Class.forName("org.sqlite.JDBC");
+            // in the tests we use sqlite, otherwise mysql is used
+            if (connectionString == null) {
+                Properties props = getConnectionProperties();
+                setConnectionString(generateConnectionString(props));
+            }
+            if (connectionString.contains("sqlite")) {
+                Class.forName("org.sqlite.JDBC");
+            } else {
+                Class.forName("org.mysql.JDBC");
+            }
+
             return DriverManager.getConnection(connectionString);
-        } catch (ClassNotFoundException e) {
+        } catch (ClassNotFoundException | IOException e) {
             throw new MigrationException("Failed to open database connection to " + connectionString, e);
         }
     }
 
     public void setConnectionString(String connectionString) {
         this.connectionString = connectionString;
     }
+
+    public void setRedirectDbConnectionPath(Path redirectDbConnectionPath) {
+        this.redirectDbConnectionPath = redirectDbConnectionPath;
+    }
+
+    public Properties getConnectionProperties() throws IOException {
+        InputStream inputStream = Files.newInputStream(redirectDbConnectionPath);
+        Properties props = new Properties();
+        props.load(inputStream);
+        return props;
+    }
+
+    public String generateConnectionString(Properties props) {
```

Comment:
Since this is a public method which does a fair amount of work, could you add javadoc to this method?